app/models/mixins/miq_request_mixin.rb

```diff
@@ -9,6 +9,20 @@ def get_option(key, value = nil)
     MiqRequestMixin.get_option(key, value, options)
   end
 
+  def get_option_decrypted(key, value = nil)
+    raise ArgumentError, "#{key} cannot be decrypted" unless option_encrypted?(key)
+    enc_value = MiqRequestMixin.get_option(password_prefixed_symbol_key(key), value, options)
+    MiqPassword.decrypt(enc_value)
+  end
```

Comment:
This seems like a duplicate of the other method.  Is there a way to DRY it up or is it too complicated?

Reply:
@Fryguy in one we directly access the options hash and in the other we access the dialog key which has another hash.